internal/authz/kessel/kessel.go

```diff
@@ -9,49 +9,85 @@ import (
 	authzapi "github.com/project-kessel/inventory-api/internal/authz/api"
 	kesselv1 "github.com/project-kessel/relations-api/api/kessel/relations/v1"
 	kessel "github.com/project-kessel/relations-api/api/kessel/relations/v1beta1"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
 	"google.golang.org/grpc"
 )
 
 type KesselAuthz struct {
-	HealthService kesselv1.KesselRelationsHealthServiceClient
-	CheckService  kessel.KesselCheckServiceClient
-	TupleService  kessel.KesselTupleServiceClient
-	tokenClient   *tokenClient
-	Logger        *log.Helper
+	HealthService  kesselv1.KesselRelationsHealthServiceClient
+	CheckService   kessel.KesselCheckServiceClient
+	TupleService   kessel.KesselTupleServiceClient
+	tokenClient    *tokenClient
+	Logger         *log.Helper
+	successCounter metric.Int64Counter
+	failureCounter metric.Int64Counter
+}
+
+var _ authzapi.Authorizer = &KesselAuthz{}
+
+func New(ctx context.Context, config CompletedConfig, logger *log.Helper) (*KesselAuthz, error) {
+	logger.Info("Using authorizer: kessel")
+	tokenCli := NewTokenClient(config.tokenConfig)
+
+	meter := otel.Meter("github.com/project-kessel/inventory-api/blob/main/internal/server/otel")
+
+	successCounter, err := meter.Int64Counter("inventory_relations_api_success")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create success counter: %v", err)
+	}
+
+	failureCounter, err := meter.Int64Counter("inventory_relations_api_failure")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create failure counter: %v", err)
+	}
+
+	return &KesselAuthz{
+		HealthService:  kesselv1.NewKesselRelationsHealthServiceClient(config.gRPCConn),
+		CheckService:   kessel.NewKesselCheckServiceClient(config.gRPCConn),
+		TupleService:   kessel.NewKesselTupleServiceClient(config.gRPCConn),
+		Logger:         logger,
+		tokenClient:    tokenCli,
+		successCounter: successCounter,
+		failureCounter: failureCounter,
+	}, nil
 }
 
 func (a *KesselAuthz) Health(ctx context.Context) (*kesselv1.GetReadyzResponse, error) {
 	opts, err := a.getCallOptions()
 	if err != nil {
+		a.failureCounter.Add(context.Background(), 1, metric.WithAttributes(attribute.String("method", "Health")))
```

Comment:
Possibly nitpicking or may not provide much value now, but since we have to frequently make calls to these counters, would it be good to make this a method of the KesselAuthz type to simplify?

```go
func (a *KesselAuthz) incrFailureCounter(method string) {
    a.failureCounter.Add(context.Background(), 1, metric.WithAttributes(attribute.String("method", method)))
}

func (a *KesselAuthz) incrSuccessCounter(method string) {
    a.successCounter.Add(context.Background(), 1, metric.WithAttributes(attribute.String("method", method)))
}
```

Then a failure or success would be
```go
a.incrFailureCounter("Health")
a.incrSuccessCounter("CreateTuple")
```
etc, etc

It may seem pointless now, but if we ever need to add more attributes other than just method, we can update it in 2 places instead of potentially 20. Thinking it might make it easier down the road if needed kinda deal. Plus if we need to make better use of the context for any reason and want to add a context to the KesselAuthz type to pass around it also makes it an easy change. Either way, this looks great to me!

Reply:
Makes sense to me, Ill include this in the next push!